docs/upgrading/upgrading-your-package.md

```diff
@@ -0,0 +1,565 @@
+
+```js
+{
+  "dependencies": {
+    "atom-space-pen-views": "^0.21"
+  }
+}
+```
+
+Then run `apm install` in your package directory.
+
+### Require views from atom-space-pen-views
+
+Anywhere you are requiring one of the following from `atom` you need to require them from `atom-space-pen-views` instead.
+
+```js
+// require these from 'atom-space-pen-views' rather than 'atom'
+$
+$$
+$$$
+View
+TextEditorView
+ScrollView
+SelectListView
+```
+
+So this:
+
+```coffee
+# Old way
+{$, TextEditorView, View, GitRepository} = require 'atom'
+```
+
+Would be replaced with this:
+
+```coffee
+# New way
+{GitRepository} = require 'atom'
+{$, TextEditorView, View} = require 'atom-space-pen-views'
+```
+
+### Run specs and test your package
+
+You wrote specs, right!? Here's where they shine. Run them with `cmd-shift-P`, and search for `run package specs`. It will show all the deprecation messages and errors.
+
+### Examples
+
+We have upgraded all the core packages. Please see [this issue](https://github.com/atom/atom/issues/4011) for a link to all the upgrade PRs.
+
+## Deprecations
+
+All of the methods in Atom core that have changes will emit deprecation messages when called. These messages are shown in two places: your **package specs**, and in **Deprecation Cop**.
+
+### Specs
+
+Just run your specs, and all the deprecations will be displayed in yellow.
+
+TODO: image of deprecations in specs
+
+TODO: Comand line spec deprecation image?
+
+### Deprecation Cop
+
+Run an atom window in dev mode (`atom -d`) with your package loaded, and open Deprecation Cop (search for `deprecation` in the command palette).
+
+TODO: image of deprecations in DepCop
+
+## Upgrading your Views
+
+Previous to 1.0, views were baked into Atom core. These views were based on jQuery and `space-pen`. They looked something like this:
+
+```coffee
+# The old way: getting views from atom
+{$, TextEditorView, View} = require 'atom'
+
+module.exports =
+class SomeView extends View
+  @content: ->
+    @div class: 'find-and-replace', =>
+      @div class: 'block', =>
+        @subview 'myEditor', new TextEditorView(mini: true)
+  #...
+```
+
+Requiring `atom` _used to_ provide the following view helpers:
+
+```
+$
+$$
+$$$
+View
+TextEditorView
+ScrollView
+SelectListView
+```
+
+### The New
+
+Atom no longer provides these view helpers baked in. Atom core is now 'view agnostic'. The preexisting view system is available from two npm packages: `space-pen`, and `atom-space-pen-views`
+
+`space-pen` now provides
+
+```
+$
+$$
+$$$
+View
+```
+
+`atom-space-pen-views` now provides all of `space-pen`, plus Atom specific views:
+
+```js
+// Passed through from space-pen
+$
+$$
+$$$
+View
+
+// Atom specific views
+TextEditorView
+ScrollView
+SelectListView
+```
+
+### Adding the module dependencies
+
+To use the new views, you need to specify the `atom-space-pen-views` module in your package's `package.json` file's dependencies:
```

Comment:
Make it clear that `atom-space-pen-views` is the dependency to include even if you only need `require 'space-pen'`?